modelscope_agent/agents_registry.py

```diff
@@ -0,0 +1,99 @@
+from typing import List, Union
+
+import ray
+from modelscope_agent.agent import Agent
+from modelscope_agent.environment import Environment
+from ray.util.client.common import ClientActorHandle
+
+
+class AgentRegistry:
+
+    def __init__(self, remote=True, **kwargs):
+        self._agents = {}
+        self._agents_state = {}
+        self.remote = remote
+
+    def register_agent(self,
+                       agent: Union[Agent, ClientActorHandle],
+                       env_context: Environment = None):
+        """
+        Add an agent to the register center
+        Args:
+            agent: Agent object
+            env_context: Env context that need to pass to agent
+        Returns: None
+
+        """
+        if isinstance(agent, Agent):
+            role = agent.role()
+        else:
+            role = ray.get(agent.role.remote())
+        self._agents[role] = agent
```

Comment:
check if role already in _agents, raise error if exists

Reply:
done